### PR TITLE
Add choice for TreasureHunt in EditSettings

### PR DESCRIPTION
**Pull request by cjordan7**

### CODE REVIEW

**Comment by cjordan7** [comment]
> It is a good first step towards location in treasure hunt quizzes. There is just one thing that I am wondering about. Treasure hunt quizzes should contain a location for every question. This means that the user shouldn't have the choice to check or uncheck the location. Will you handle that in a future PR, or do you plan on doing it now ?

Yeah, I think I will change it. I will rename location to TreasureHunt checkbox and the user will have to give a location

File: app/src/main/java/ch/epfl/qedit/model/Quiz.java [hunk 1]
```diff
@@ -65,6 +65,11 @@ public Builder(Quiz quiz) {
             this.treasureHunt = quiz.treasureHunt;
         }
 
+        public Builder setTreasureHunt(Boolean isTreasureHunt) {
```
Comment (brunnerant) [inline app/src/main/java/ch/epfl/qedit/model/Quiz.java:68]: I don't think it is a really good idea to have that method here. That means that you can change the treasure hunt boolean in the middle of building a quiz, meaning that you could make the questions invalid.

File: app/src/main/res/layout/activity_edit_modify_quiz_settings.xml [hunk 2]
```diff
@@ -21,6 +21,19 @@
         app:layout_constraintBottom_toTopOf="@+id/button_start_editing"
         android:inputType="text" />
 
+    <CheckBox
+        android:id="@+id/treasure_hunt_checkbox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/regular_padding"
+        android:text="@string/treasureHunt"
```
Comment (brunnerant) [inline app/src/main/res/layout/activity_edit_modify_quiz_settings.xml:29]: I think that the convention is to use `snake_case` in the xml files.

File: app/src/main/res/layout/activity_edit_new_quiz_settings.xml [hunk 3]
```diff
@@ -21,6 +21,19 @@
         app:layout_constraintBottom_toTopOf="@+id/edit_language_selection"
         android:inputType="text" />
 
+    <CheckBox
+        android:id="@+id/treasure_hunt_checkbox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/regular_padding"
+        android:text="@string/treasureHunt"
```
Comment (brunnerant) [inline app/src/main/res/layout/activity_edit_new_quiz_settings.xml:29]: Ssssnake cassssse.

File: app/src/main/res/layout/activity_edit_modify_quiz_settings.xml [hunk 4]
```diff
@@ -21,6 +21,19 @@
         app:layout_constraintBottom_toTopOf="@+id/button_start_editing"
         android:inputType="text" />
 
+    <CheckBox
```
Comment (brunnerant) [inline app/src/main/res/layout/activity_edit_modify_quiz_settings.xml:24]: How do you handle the case when a non-treasure hunt is converted to a treasure hunt quiz ? It means that the user must add a location to all the questions. I think that the easiest thing for now is to not allow the user to change that setting.

**Pull request merged**